modular/blocksyncer/blocksyncer_indexer.go

```diff
@@ -1,551 +0,0 @@
-}
-
-func (i *Impl) BatchHandle(ctx context.Context, dataList []interface{}, block *coretypes.ResultBlock, txHash common.Hash) error {
-	objects := make([]*models.Object, 0)
-	objectIDMap := make(map[common.Hash]bool, 0)
-	bucketNameMap := make(map[string][]common.Hash)
-	streamRecords := make([]*models.StreamRecord, 0)
-	paymentAccount := make([]*models.PaymentAccount, 0)
-
-	for _, data := range dataList {
-		switch m := data.(type) {
-		case *models.Object:
-			objects = append(objects, m)
-			objectIDMap[m.ObjectID] = true
-			tmp := bucketNameMap[m.BucketName]
-			if tmp == nil {
-				tmp = make([]common.Hash, 0)
-			}
-			tmp = append(tmp, m.ObjectID)
-			bucketNameMap[m.BucketName] = tmp
-		case *models.StreamRecord:
-			streamRecords = append(streamRecords, m)
-		case *models.PaymentAccount:
-			paymentAccount = append(paymentAccount, m)
-		}
-	}
-
-	localDB := localdb.Cast(i.DB)
-	if len(paymentAccount) > 0 {
-		if err := localDB.BatchSavePaymentAccount(ctx, paymentAccount); err != nil {
-			return err
-		}
-	}
-	if len(streamRecords) > 0 {
-		if err := localDB.BatchSaveStreamRecord(ctx, streamRecords); err != nil {
-			return err
-		}
-	}
-
-	insert := make(map[string][]*models.Object)
-	update := make(map[string][]*models.Object)
-
-	objResults, err := localDB.GetObjectList(ctx, bucketNameMap)
-	if err != nil {
-		return err
-	}
-	resMap := make(map[common.Hash]*models.Object)
-	for _, obj := range objResults {
-		resMap[obj.ObjectID] = obj
-	}
-
-	for _, obj := range objects {
-		res, ok := resMap[obj.ObjectID]
-		if !ok {
-			tmp := insert[obj.BucketName]
-			if tmp == nil {
-				tmp = make([]*models.Object, 0)
-			}
-			tmp = append(tmp, obj)
-			insert[obj.BucketName] = tmp
-			continue
-		}
-		if obj.Removed {
-			res.Removed = true
-			res.BucketName = obj.BucketName
-			res.ObjectName = obj.ObjectName
-			res.ObjectID = obj.ObjectID
-			res.LocalVirtualGroupId = obj.LocalVirtualGroupId
-			res.UpdateAt = block.Block.Height
-			res.UpdateTxHash = txHash
-			res.UpdateTime = time.Now().UTC().Unix()
-		}
-		resMap[obj.ObjectID] = res
-	}
-
-	if len(insert) > 0 {
-		if err := localDB.BatchSaveObject(ctx, insert); err != nil {
-			return err
-		}
-	}
-	for _, v := range resMap {
-		tmp := update[v.BucketName]
-		if tmp == nil {
-			tmp = make([]*models.Object, 0)
-		}
-		tmp = append(tmp, v)
-		update[v.BucketName] = tmp
-	}
-	if len(update) > 0 {
-		if err := localDB.BatchSaveObject(ctx, update); err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
-func (i *Impl) concurrenceHandleEvent(ctx context.Context, block *coretypes.ResultBlock, allEvents [][]TxHashEvent) error {
-	wg := &sync.WaitGroup{}
-	wg.Add(len(allEvents))
-	var handleErr error
-	for _, events := range allEvents {
-		go func(event []TxHashEvent) {
-			defer wg.Done()
-			for _, e := range event {
-				if err := i.HandleEvent(ctx, block, e.TxHash, e.Event); err != nil {
-					log.Errorf("failed to HandleEvent err:%v", err)
-					handleErr = err
```

Comment:
concurrent write to `handleErr`?

Reply:
Version issue. This code has been removed